### PR TITLE
Button bug

### PR DESCRIPTION
**Pull request by andrea-grapl**
### Which issue does this PR correspond to?
https://github.com/grapl-security/issue-tracker/issues/903

### What changes does this PR make to Grapl? Why?

Remove directional arrows & link particles - distracting 
Lens Names are all capped, should be case sensitive - Better UX
Highlight Current Page on Nav Bar - Better UX
Ability to copy and paste lens names "click here to copy lens name" feature - Better UX

### How were these changes tested?

Manually - `make test-e2e`, click sidebar, lens names, copy lens names, etc.

### CODE REVIEW

File: src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx [hunk 1]
```diff
@@ -187,29 +184,29 @@ const GraphDisplay = ({ lensName, setCurNode }: GraphDisplayProps) => {
...
+            // ctx.save();
             ctx.beginPath();
             ctx.arc(node.x, node.y, NODE_R * 1.4, 0, 2 * Math.PI, false);
             ctx.fillStyle =
                 node === hoverNode
                     ? colors.hoverNodeFill
                     : riskOutline(node.risk_score);
             ctx.fill();
-            ctx.restore();
+            // ctx.restore();
```
Comment (colin-grapl) [inline src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx:198]: let's remove these commented lines

File: src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx [hunk 2]
```diff
@@ -187,29 +184,29 @@ const GraphDisplay = ({ lensName, setCurNode }: GraphDisplayProps) => {
...
 
             // Node Fill Styling
-            ctx.save();
+            // ctx.save();
```
Comment (colin-grapl) [inline src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx:201]: delete

File: src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx [hunk 3]
```diff
@@ -218,7 +215,7 @@ const GraphDisplay = ({ lensName, setCurNode }: GraphDisplayProps) => {
                 98,
                 NODE_R / ctx.measureText(label).width
             );
-            ctx.save();
+            // ctx.save();
```
Comment (colin-grapl) [inline src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx:218]: delete

File: src/js/engagement_view/src/types/LensAndNodeTableTypes.tsx [hunk 4]
```diff
@@ -6,6 +6,8 @@ export type SelectLensProps = {
     uid: number;
     lens_type: string;
     setLens: (lens: string) => void;
+    selectedId: number;
+    setLensTableState: any;
```
Comment (colin-grapl) [inline src/js/engagement_view/src/types/LensAndNodeTableTypes.tsx:10]: let's type this instead of any

File: src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTable.tsx [hunk 5]
```diff
@@ -15,6 +15,8 @@ import { PaginationState } from "types/CustomTypes";
 
 export const lensTable = (
     state: PaginationState,
+    selectedIdState: number,
+    setSelectedIdState: (selectedIdState: number) => void,
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTable.tsx:19]: preface for the rest of this PR: I am not a react expert

I guess I'm confused why there's additional state outside the variable named `state`
could we rename that to `paginationState`?

File: src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTable.tsx [hunk 6]
```diff
@@ -71,6 +73,8 @@ export const lensTable = (
                                     lens_type={lens.lens_type}
                                     score={lens.score}
                                     setLens={setLens}
+                                    selectedId={selectedIdState}
+                                    setLensTableState={setSelectedIdState}
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTable.tsx:77]: there's an incongruency in these args and their values, could we maybe sync them up a bit more if it makes sense to do so?

File: src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTableStyles.tsx [hunk 7]
```diff
@@ -56,7 +56,12 @@ export const useStyles = makeStyles((theme) =>
             width: "95%",
         },
         tableRow: {
-            background: theme.palette.background.default,
+            "&.Mui-selected": {
+                backgroundColor: "#2B3648",
+                "& > .MuiTableCell-root": {
+                    color: "#1976D2",
+                },
+            },
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTableStyles.tsx:64]: It's a bit odd to me that these are the only bare hex-colors in this file. Is there any way we can associate them with the Theme palette? Would that make sense to do or are they unrelated?

File: src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/selectLens.tsx [hunk 8]
```diff
@@ -11,21 +10,19 @@ import { SelectLensProps } from "types/LensAndNodeTableTypes";
...
-                    // className={classes.lensName}
-                    className={btnClass ? "btnClass clicked" : "btnClass"}
-                    onClick={() => {
-                        props.setLens(props.lens);
-                        btnClass ? setBtnClass(false) : setBtnClass(true);
-                    }}
-                >
-                    {props.lens_type + " :\t\t" + props.lens}
-                </Button>
+                {props.lens_type + " :\t\t" + props.lens}
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/selectLens.tsx:25]: could you leave a comment on what does :\t\t do

File: src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTable.tsx (hunk 6 above)
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTable.tsx:77]: could we call this setLensSelectedId or something?

File: src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx [hunk 9]
```diff
@@ -187,29 +184,25 @@ const GraphDisplay = ({ lensName, setCurNode }: GraphDisplayProps) => {
...
-            ctx.save();
             ctx.beginPath();
             ctx.arc(node.x, node.y, NODE_R * 1.4, 0, 2 * Math.PI, false);
             ctx.fillStyle =
                 node === hoverNode
                     ? colors.hoverNodeFill
                     : riskOutline(node.risk_score);
             ctx.fill();
-            ctx.restore();
+            // ctx.restore();
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx:196]: I see this as 'resolved' but it's still commented?

File: src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx [hunk 10]
```diff
@@ -249,6 +241,10 @@ const GraphDisplay = ({ lensName, setCurNode }: GraphDisplayProps) => {
         const LABEL_NODE_MARGIN = 12;
         const start = link.source;
         const end = link.target;
+
+        link.fx = link.x;
+        link.fy = link.y;
+
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx:247]: # try this change after you do everything else
and i'm not opposed if you do it in a followup PR, as long as you do _do_ it 
---

We can actually get correct types on `constlinkStyling` fwiw
https://github.com/vasturiano/react-force-graph/blob/master/src/packages/react-force-graph-2d/index.d.ts#L71
```
it's a CanvasCustomRenderFn<LinkObject>;
which boils down to a 
(link: LinkObject, canvasContext: CanvasRenderingContext2D) => {
```

you'll have to import LinkObject type from react-force-graph-2d
and you'll have to import CanvasRenderingContext2D from some presupplied web api typescript gives you
https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D

---
that said, here I see you doing like `link.x, link.y` whereas LinkObject in the linked source code says it only has properties `.source` and `.target`, so, I'm a tad confused how that all works...

File: src/js/engagement_view/src/components/reusableComponents/collapsibleDrawer.tsx [hunk 11]
```diff
@@ -107,40 +116,56 @@ export default function CollapsibleNavDrawer() {
                 </DrawerHeader>
                 <Divider />
                 <List className={classes.drawer}>
-                    <ListItem button key="Analyzers">
+                    <ListItemButton
+                        key="Analyzers"
+                        selected={selectedIndex === 0}
+                        onClick={(event) => handleListItemClick(event, 0)}
+                    >
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/reusableComponents/collapsibleDrawer.tsx:123]: I won't stop the PR on this, but this section definitely has a code smell. Consider doing something like the following:
```
interface ListItem {
  label: String,
  link: String,
}

const items: ListItem[] = [
  { label: "Analyzers", link: "/analyzers" },
  { label: "Engagements", link: "/engagements" },
];

items.forEach((value, index) => {
	// render each item as a ListItemButton
});

```

File: src/js/engagement_view/src/components/reusableComponents/drawer.tsx [hunk 12]
```diff
@@ -34,6 +35,21 @@ const DrawerHeader = styled("div")(({ theme }) => ({
 export const NavigationDrawer = () => {
     const classes = useStyles();
 
+    const location = window.location.href.split("/").slice().pop();
+
+    let index: number = 0;
+
+    location === "analyzers" ? (index = 0) : (index = 2);
+
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/reusableComponents/drawer.tsx:43]: Okay, this I'd block the PR on. It's basically re-stating the "analyzers is 0" mapping in collapsibleDrawer.tsx, but obviously the code here is wrong (it's not always 2 if it's not analyzers)

Comment (wimax-grapl) [reply src/js/engagement_view/src/components/reusableComponents/drawer.tsx:43]: While I don't love using the window.location as a tracker of state, you could use it and the aforementioned 
```

const items: ListItem[] = [
  { label: "Analyzers", link: "/analyzers" },
  { label: "Engagements", link: "/engagements" },
];
```
to accurately figure out which index it currently is.

File: src/js/engagement_view/src/components/reusableComponents/drawer.tsx [hunk 13]
```diff
@@ -56,40 +72,55 @@ export const NavigationDrawer = () => {
             </DrawerHeader>
 
             <List className={classes.drawer}>
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/reusableComponents/drawer.tsx:74]: More proof of repeated code-smell - this entire section is repeated across files. Maybe this can be extracted out to its own component?

**Review by wimax-grapl — CHANGES_REQUESTED** [review]

File: src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTable.tsx (hunk 5 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTable.tsx:19]: Sorry, this is unclear - this state refers to the selected lens, not selected pagination. We have two separate states here, one for pagination, one for current selected lens. Will rename the variables to be more clear.

File: src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx (hunk 10 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx:247]: This is dead code that shouldn't be in here, sorry - was testing something and forgot to remove this.You're right with the  source and target.

File: src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTableStyles.tsx (hunk 7 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTableStyles.tsx:64]: Moved to theme, now referenced in pallete

File: src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/selectLens.tsx (hunk 8 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/selectLens.tsx:25]: Dead and unnecessary code - removed.

File: src/js/engagement_view/src/components/reusableComponents/collapsibleDrawer.tsx (hunk 11 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/components/reusableComponents/collapsibleDrawer.tsx]: Used a new feature to set active links in upgrade to react router dependency.

File: src/js/engagement_view/src/components/reusableComponents/drawer.tsx (hunk 13 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/components/reusableComponents/drawer.tsx]: Also resolved with react router update!

File: src/js/engagement_view/src/components/reusableComponents/drawer.tsx (hunk 12 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/components/reusableComponents/drawer.tsx]: Set active link with upgrade to react router!

File: src/js/engagement_view/Makefile [hunk 14]
```diff
@@ -108,7 +108,7 @@ shell: ## Drop into a shell inside the build container
 
 .PHONY: start
 start: extra_docker_run_args := --interactive --tty --network host
-start: build-image
+start: build-code
```
Comment (wimax-grapl) [inline src/js/engagement_view/Makefile:111]: This suggests that you need to `yarn build` (make a release build) before you `yarn start` (which is the debugging one that auto-updates I believe), which seems incorrect.

File: src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTable.tsx [hunk 15]
```diff
@@ -71,6 +73,8 @@ export const lensTable = (
                                     lens_type={lens.lens_type}
                                     score={lens.score}
                                     setLens={setLens}
+                                    selectedId={selectedIdState}
+                                    setSelectedLens={setSelectedIdState}
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTable.tsx:77]: There's a mismatch in these state and setter names.

IMO this should look like
```
                                    selectedLensId={selectedLensIdState}
                                    setSelectedLensId={setSelectedLensIdState}
```

a `number` named `id` tells me nothing about how that variable/state is used; I almost always try to label my IDs (lens_id, page_id, user_id, etc).

File: src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/selectLens.tsx [hunk 16]
```diff
@@ -11,21 +10,19 @@ import { SelectLensProps } from "types/LensAndNodeTableTypes";
...
-    const [btnClass, setBtnClass] = useState(false);
-
     return (
-        <TableRow className={classes.tableRow} key={props.uid}>
+        <TableRow
+            hover
+            key={props.uid}
+            onClick={() => {
+                props.setSelectedLens(props.uid);
+                props.setLens(props.lens);
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/selectLens.tsx:19]: more proof that, yeah, setSelectedLens should be named setSelectedLensId or setSelectedLensUid or something - the fact that `setSelectedLens` and `setLens` take two different types isn't great.

File: src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx [hunk 17]
```diff
@@ -218,7 +210,6 @@ const GraphDisplay = ({ lensName, setCurNode }: GraphDisplayProps) => {
                 98,
                 NODE_R / ctx.measureText(label).width
             );
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx:212]: and just to make sure, all this removal of ctx is expected?

File: src/js/engagement_view/src/components/reusableComponents/collapsibleDrawer.tsx [hunk 18]
```diff
@@ -101,59 +91,17 @@ export default function CollapsibleNavDrawer() {
                             )}
                         </IconButton>
                         <Icon>
-                            <img className={classes.logoImage} src={Img} />
+                            <img
+                                className={classes.logoImage}
+                                src={Img}
+                                alt="Grapl Logo"
+                            />
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/reusableComponents/collapsibleDrawer.tsx:98]: Looks like we have a number of these - perhaps we should make this its own `<GraplLogo />` component?

image

File: src/js/engagement_view/src/components/reusableComponents/drawerList.tsx [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+        <>
+            <List className={classes.drawer}>
+                <ListItemButton>
+                    <NavLink
+                        to="/analyzers"
+                        className={classes.navLink}
+                        style={({ isActive }) => ({
+                            color: isActive ? "#1A76D2" : "#C6D1E7",
+                        })}
+                    >
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/reusableComponents/drawerList.tsx:29]: It looks like react-router offers a more robust way to do this:

https://v5.reactrouter.com/web/api/NavLink/activeclassname-string

by default, when a NavLink becomes active, its `class` becomes `className + ".active"`

so I think you can just do this in the styles file

```
        navLink: {
            textDecoration: "none",
            color: theme.palette.info.contrastText,
            "& .active": {
              color: theActiveColorGoesHere
            }
        },
```

I _believe_ the & is how you do `.navlink.active` in this CSS library, but I'm not entirely sure.

File: src/js/engagement_view/src/theme.tsx [hunk 20]
```diff
@@ -30,6 +31,18 @@ const theme = createTheme({
             contrastText: "#FFFFFF",
         },
     },
+    components: {
+        MuiListItemButton: {
+            styleOverrides: {
+                root: {
+                    "&.Mui-selected": {
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/theme.tsx:38]: yeah, this is the same `& .class` thing I mentioned above, for the record

File: src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx (hunk 17 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx:212]: Yes

File: src/js/engagement_view/Makefile (hunk 14 above)
Comment (andrea-grapl) [reply src/js/engagement_view/Makefile:111]: Leaving per cm

File: src/js/engagement_view/src/components/reusableComponents/collapsibleDrawer.tsx (hunk 18 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/components/reusableComponents/collapsibleDrawer.tsx:98]: Done. Good call.

File: src/js/engagement_view/src/components/reusableComponents/drawerList.tsx (hunk 19 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/components/reusableComponents/drawerList.tsx:29]: I think the above method was right for v5 of react router, but according to the linked docs for v6 (which we just started using) I think the docs want us to style active links with functions. Happy to change this, but according to the following paragraph, I think we're supposed to be using inline styles, but could totally be reading this wrong. Happy to change if necessary. 

```
In React Router v6, activeStyle will be removed and you should use the function style to apply inline styles to either active or inactive NavLink components.
```

File: src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/selectLens.tsx (hunk 16 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/selectLens.tsx]: done

File: src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTable.tsx (hunk 15 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTable.tsx]: done

File: src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTable.tsx (hunk 6 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/components/engagementView/sidebar/tables/lensTable/lensTable.tsx]: resolved in different comment

**Review by wimax-grapl — APPROVED** [review]
Approved, and I won't stop the PR if you don't include this, but worth noting that the `.active` suggestion still exists.

per slack:
```
actually, after looking at the v6 docs, it looks like my suggestion still stands
https://reactrouter.com/docs/en/v6/api#navlink
By default, an active class is added to a <NavLink> component when it is active. This provides the same simple styling mechanism for most users who are upgrading from v5. One difference as of v6.0.0-beta.3 is that activeClassName and activeStyle have been removed from NavLinkProps.
basically v5 gave you too many options to customize stuff so they ripped it out in v6

```
